src/integration/appland/upload.ts

```diff
@@ -1,85 +1,109 @@
-import { pack } from 'tar-stream';
 import { promises as fs } from 'fs';
-import { createHash } from 'crypto';
-import FormData from 'form-data';
-import { createGzip } from 'zlib';
-import { buildAppMap } from '@appland/models';
-import { buildRequest, handleError } from '@appland/client/dist/src';
-import { ScanResults } from '../../report/scanResults';
 import { IncomingMessage } from 'http';
 import { URL } from 'url';
+import { queue } from 'async';
 
-export default async function (scanResults: ScanResults, appId: string): Promise<void> {
-  const normalizedFilePaths: { [key: string]: string } = {};
-  const { findings } = scanResults;
-  for (const finding of findings) {
-    if (!finding.appMapFile) {
-      continue;
-    }
+import { AppMap as AppMapStruct } from '@appland/models';
+import { buildRequest, handleError } from '@appland/client/dist/src';
 
-    const hash = createHash('sha256').update(finding.appMapFile).digest('hex');
-    normalizedFilePaths[finding.appMapFile] = `${hash}.appmap.json`;
-  }
+import { ScanResults } from '../../report/scanResults';
+import { AppMap as AppMapClient } from './appMap';
+import { Mapset as MapsetClient } from './mapset';
+import { join } from 'path';
 
-  const clonedFindings = findings.map((finding) => {
-    const clone = { ...finding };
-    if (clone.appMapFile) {
-      clone.appMapFile = normalizedFilePaths[clone.appMapFile];
-    }
-    return clone;
-  });
+export default async function (
+  scanResults: ScanResults,
+  appId: string,
+  appmapDir: string
+): Promise<URL> {
+  console.warn(`Uploading AppMaps and findings to application '${appId}'`);
+
+  const { findings } = scanResults;
 
   const relevantFilePaths = [
     ...new Set(findings.filter((f) => f.appMapFile).map((f) => f.appMapFile)),
   ] as string[];
-  const tarStream = pack();
 
-  for (const filePath of relevantFilePaths) {
-    const buffer = await fs.readFile(filePath);
+  const appMapUUIDByFileName: Record<string, string> = {};
+  const branchCount: Record<string, number> = {};
+  const commitCount: Record<string, number> = {};
 
-    tarStream.entry(
-      { name: normalizedFilePaths[filePath] },
-      JSON.stringify(buildAppMap(buffer.toString()).normalize().build().toJSON())
-    );
-  }
+  const q = queue(async (filePath: string, callback) => {
+    console.log(`Uploading AppMap ${filePath}`);
+    const buffer = await fs.readFile(join(appmapDir, filePath));
+    const appMapStruct = JSON.parse(buffer.toString()) as AppMapStruct;
+    const branch = appMapStruct.metadata.git?.branch;
+    const commit = appMapStruct.metadata.git?.commit;
+    if (branch) {
+      branchCount[branch] ||= 1;
+      branchCount[branch] += 1;
+    }
+    if (commit) {
+      commitCount[commit] ||= 1;
+      commitCount[commit] += 1;
+    }
```

Comment:
Asking to better understand the flow. What is the use case when someone tries uploading appmaps from different branches?